src/bin/cargo-miri.rs

```diff
@@ -294,6 +294,7 @@ fn setup(ask_user: bool) {
     // The interesting bit: Xargo.toml
     File::create(dir.join("Xargo.toml")).unwrap()
         .write_all(br#"
+cargo_mode = "check"
```

Comment:
Ah, making xargo do a check-only build is something I also wanted to try for a while. ;)